modules/module_manager/test/modulemanager_test.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+/**
+ * Module Manager automated integration tests
+ *
+ * PHP Version 7
+ *
+ * @category Test
+ * @package  Loris
+ * @author   Wang Shen <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+require_once __DIR__ .
+    "/../../../test/integrationtests/LorisIntegrationTest.class.inc";
+/**
+ * Module Manager automated integration tests
+ *
+ * @category Test
+ * @package  Loris
+ * @author   Wang Shen <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class ModuleManagerTest extends LorisIntegrationTest
+{
+
+    /**
+     * Tests that the module_manager loads an error page if
+     * the user has no permission.
+     *
+     * @return void
+     */
+    function testLoadPageWithoutPermission()
+    {
+        $this->setupPermissions(array(""));
+        $this->safeGet($this->url . "/module_manager/");
+
+        // Test that the Imaging menu appears in the first row
+        $bodyText = $this->webDriver->findElement(
+            WebDriverBy::cssSelector("body")
+        )->getText();
+        $this->assertContains(
+            "You do not have access to this page.",
```

Comment:
Maybe this string could be a class constant. It's a nit-pick but if this module is going to serve as a clean example for future work then it would be nice.

Reply:
1. I disagree that that's cleaner.
2. I'll clarify at the LORIS meeting next week that this point of this module is *not* to be a clean example. It will evolve in ways that make it a poor example once more complex/module specific functionality gets added. It's just that, at the moment, it's one of the cleanest examples.